apps/chat/src/store/prompts/prompts.epics.ts

```diff
@@ -933,12 +933,32 @@ const uploadPromptsWithFoldersEpic: AppEpic = (action$) =>
         switchMap((foldersAndEntities) => {
           const folders = foldersAndEntities.flatMap((f) => f.folders);
           const prompts = foldersAndEntities.flatMap((f) => f.entities);
-          return of(
-            PromptsActions.uploadChildPromptsWithFoldersSuccess({
-              parentIds: payload.ids,
-              folders,
-              prompts,
-            }),
+          return concat(
+            of(
+              PromptsActions.uploadChildPromptsWithFoldersSuccess({
+                parentIds: payload.ids,
+                folders,
+                prompts,
+              }),
+            ),
+            iif(
+              () => !!payload.selectFirst,
+              concat(
+                of(
+                  PromptsActions.setSelectedPrompt({
+                    promptId: prompts[0].id,
+                  }),
+                ),
+                of(PromptsActions.uploadPrompt({ promptId: prompts[0].id })),
+                of(
+                  UIActions.openFolder({
+                    featureType: FeatureType.Prompt,
+                    id: prompts[0].folderId,
```

Comment:
Did you check that `prompts.length > 0`?
```suggestion
              () => !!payload.selectFirst && prompts.length > 0,
              concat(
                of(
                  PromptsActions.setSelectedPrompt({
                    promptId: prompts[0].id,
                  }),
                ),
                of(PromptsActions.uploadPrompt({ promptId: prompts[0].id })),
                of(
                  UIActions.openFolder({
                    featureType: FeatureType.Prompt,
                    id: prompts[0].folderId,
```